pkg/controller/kieapp/defaults/auth.go

```diff
@@ -0,0 +1,93 @@
+package defaults
+
+import (
+	"fmt"
+
+	"github.com/kiegroup/kie-cloud-operator/pkg/apis/app/v1"
+	"github.com/kiegroup/kie-cloud-operator/pkg/controller/kieapp/constants"
+	"github.com/kiegroup/kie-cloud-operator/pkg/controller/kieapp/shared"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const ssoHostnameVar = "HOSTNAME_HTTPS"
+const ssoClientVar = "SSO_CLIENT"
+
+// ConfigureHostname sets the HOSTNAME_HTTPS environment variable with the provided hostname
+// IF not yet set AND SSO auth is configured AND SSO_CLIENT exists
+func ConfigureHostname(object *v1.CustomObject, cr *v1.KieApp, hostname string) {
+	if cr.Spec.Auth.Type != v1.SSOAuthType {
+		return
+	}
+	for dcIdx := range object.DeploymentConfigs {
+		dc := &object.DeploymentConfigs[dcIdx]
+		for containerIdx := range dc.Spec.Template.Spec.Containers {
+			container := &dc.Spec.Template.Spec.Containers[containerIdx]
+			if pos := shared.GetEnvVar(ssoClientVar, container.Env); pos == -1 {
+				continue
+			}
+			if pos := shared.GetEnvVar(ssoHostnameVar, container.Env); pos == -1 {
+				container.Env = append(container.Env, corev1.EnvVar{
+					Name:  ssoHostnameVar,
+					Value: hostname,
+				})
+			} else if len(container.Env[pos].Value) == 0 {
+				container.Env[pos].Value = hostname
+			}
+		}
+	}
+}
+
+func configureAuth(spec v1.KieAppSpec, envTemplate *v1.EnvTemplate) error {
+	var err error
+	if len(spec.Auth.Type) == 0 {
+		return nil
+	}
+	if spec.Auth.Type == v1.SSOAuthType {
+		err = configureSSO(spec.Auth.SSO, spec.KieDeployments, envTemplate)
+	} else if spec.Auth.Type == v1.LDAPAuthType {
+		err = configureLDAP(spec.Auth.LDAP, envTemplate)
+	} else {
+		return fmt.Errorf("invalid authentication type %v", spec.Auth.Type)
+	}
+	if spec.Auth.RoleMapper != nil {
+		err = configureRoleMapper(spec.Auth.RoleMapper, envTemplate)
+	}
+	return err
+}
+
+func configureSSO(config *v1.SSOAuthConfig, deployments int, envTemplate *v1.EnvTemplate) error {
+	if config == nil {
+		return errors.New("missing SSO configuration")
+	}
+	if len(config.URL) == 0 || len(config.Realm) == 0 {
+		return errors.New("neither url nor realm can be emtpy")
+	}
+	// Set defaults
+	if len(config.PrincipalAttribute) == 0 {
+		config.PrincipalAttribute = constants.SSODefaultPrincipalAttribute
+	}
+	if len(config.DisableSSLCertValidation) == 0 {
+		config.DisableSSLCertValidation = constants.SSODefaultDisableSSLCertValidation
+	}
+	envTemplate.Auth.SSO = *config.DeepCopy()
+	return nil
+}
+
+func configureLDAP(config *v1.LDAPAuthConfig, envTemplate *v1.EnvTemplate) error {
+	if config == nil {
+		return errors.New("missing LDAP configuration")
+	}
+	if len(config.URL) == 0 {
+		return errors.New("the url must not be empty")
+	}
+	envTemplate.Auth.LDAP = *config.DeepCopy()
+	return nil
+}
+
+func configureRoleMapper(config *v1.RoleMapperAuthConfig, envTemplate *v1.EnvTemplate) error {
```

Comment:
Are you just trying to standardize on returning an error? Not sure if that's worth it. Result is that you have an extra check on the calling function, or in your case above, you overwrite a potential error with nil.

Reply:
Fixing